src/compiler/types.ts

```diff
@@ -689,7 +689,7 @@ module ts {
         hasExtendedUnicodeEscape?: boolean;
     }
 
-    export interface StringLiteralExpression extends LiteralExpression {
+    export interface StringLiteralExpression extends LiteralExpression, TypeNode {

```

Comment:
Just a naming issue, but I feel like this should be StringLiteral more than StringLiteralExpression. If it's a type node, it's not an expression